packages/@aws-cdk/aws-lambda-nodejs/lib/inline-function.ts

```diff
@@ -0,0 +1,61 @@
+import * as lambda from '@aws-cdk/aws-lambda';
+import * as cdk from '@aws-cdk/core';
+
+/**
+ * Properties for a NodejsInlineFunction
+ */
+export interface INodejsInlineFunctionProps {
```

Comment:
Is there a way to `extends lambda.FunctionOptions` here? I get `error TS9999: JSII: Attempted to extend struct @aws-cdk/aws-lambda.FunctionOptions from regular interface @aws-cdk/aws-lambda-nodejs.INodejsInlineFunctionProps` and if make it a `NodejsInlineFunctionProps` I get `error TS9999: JSII: Interface contains behavior: name should be "INodejsInlineFunctionProps"`

Reply:
@jogold from my understanding, the issue here isn't what the errors suggest.
You have a function type in this props interface, which JSII doesn't allow. 
This is because JSII doesn't (yet) know how to translate function types on interfaces to all the supported languages.

Some possible solutions could be to:
- either let the users serialize their functions themselves, & pass in a string instead
- or, look into adding support for more complex types in JSII, including "free-floating" function types

But, those are my guesses.
@RomainMuller or @eladb might be able to shine more light on the issue.